root/static/scripts/common/MB/Control/Autocomplete.js

```diff
@@ -466,9 +472,18 @@ $.widget('mb.entitylookup', $.ui.autocomplete, {
       action: this._searchAgain.bind(this, true),
     });
 
-    const allowCreation = window === window.top;
     const entity = this.entity.replace('-', '_');
 
+    const isTopWindow = window === window.top;
+    const user = window.__MB__?.$c.user;
+    let userCanAdd = true;
+    if (entity === 'area') {
+      userCanAdd = isLocationEditor(user);
+    } else if (entity === 'instrument') {
+      userCanAdd = isRelationshipEditor(user);
+    }
+    const allowCreation = isTopWindow && userCanAdd;
```

Comment:
Approving though I wouldn't both calculating `userCanAdd` if we already know `isTopWindow` is false.
```suggestion
    const user = window.__MB__?.$c.user;
    let userCanAdd = false;
    if (isTopWindow) {
      if (entity === 'area') {
        userCanAdd = isLocationEditor(user);
      } else if (entity === 'instrument') {
        userCanAdd = isRelationshipEditor(user);
      } else {
        userCanAdd = true;
      }
    }
    const allowCreation = isTopWindow && userCanAdd;
```

Reply:
If we're already having `userCanAdd` be `false` unless `isTopWindow`, should we even check `isTopWindow` again in `allowCreation`? We could just check `userCanAdd` directly and save one check, I guess?

Reply:
Yeah, good point. You only need one variable, then, 'cause `const allowCreation = userCanAdd;` is also pointless in that case.